### PR TITLE
Bug 1214569 - [User Story] Unpin page from pin dialog

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4a0f19ed554520d87723df5bada2eaef54c78236

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e36e0caf16436add3b8460418ad99e4762daf205

File: apps/system/js/pin_page_system_dialog.js [hunk 1]
```diff
@@ -166,6 +167,17 @@
     siteBadge.icon = data.icon;
     this.siteName.textContent = data.name;
 
+    Service.request('Places:isPinned', data.url)
+      .then((isPinned) => {
+        if (isPinned) {
```
Comment (albertopq) [inline apps/system/js/pin_page_system_dialog.js]: What about:

```javascript
var action = isPinned ? 'unpin' : 'pin';
this.pinPageButton.setAttribute('data-l10n-id', 'pinning-' + action + '-page');
this.pinPageButton.dataset.action = action + '-page';
```
I think is more readable, but feel free to leave as it is if you prefer

Comment (gmarty) [inline apps/system/js/pin_page_system_dialog.js]: Yes I agree, but I just took the same syntax than pinning site just a few lines below. So leaving as is for now.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ca3be780fe2b9c308ce0886b77c788e4bc2b3135

File: apps/system/js/pin_page_system_dialog.js (hunk 1 above)
Comment (albertopq) [inline apps/system/js/pin_page_system_dialog.js]: I would prefer changing both, but ok!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ddb13bfca1b25f19f8c03b92dda4a099955c5f01

**Pull request merged**